ansible_base/rbac/validators.py

```diff
@@ -116,6 +116,16 @@ def validate_permissions_for_model(permissions, content_type: Optional[Model], m
                 display_perms = ', '.join(non_add_model_permissions)
                 raise ValidationError({'permissions': f'Permissions for model {role_model._meta.verbose_name} needs to include view, got: {display_perms}'})
 
+    # check change edit permissions is given for every model that has update/delete/special actions listed
+    for cls, valid_model_permissions in permissions_by_model.items():
+        if 'delete' in cls._meta.default_permissions:
+            # import pdb; pdb.set_trace()
+            model_permissions = set(valid_model_permissions) & codename_list
+            non_add_model_permissions = {codename for codename in model_permissions if not is_add_perm(codename)}
+            if 'delete' in non_add_model_permissions and not any('change' in codename for codename in non_add_model_permissions):
```

Comment:
I don't think `'delete' in non_add_model_permissions` works, because in the case of credential, for example, the permission codename is "delete_credential".

And I'll volunteer that the current code is inconsistent and messy, in that `is_add_perm` is a method, but we should really have a similar method for any permission that is ever used in internal policy or validation logic. And it would probably be best to put that on the `DABPermission` model. I just haven't gotten around to doing that.

But in any case, I expect this condition to do a test "is this a delete codename" for any element in the list.